src/scripts/actions/UserActions.js

```diff
@@ -88,17 +92,17 @@ export const setupKeystore = () => async (
       walletIsValid = false
     }
   }
-  if (!wallet || !walletIsValid) {
+  if (!decryptedWallet || !walletIsValid) {
     paratii.eth.wallet.create()
     mnemonic = await paratii.eth.wallet.getMnemonic()
-    wallet = paratii.eth.wallet.encrypt(DEFAULT_PASSWORD)
-    dispatch(
-      setAndSyncWalletData({
-        wallet,
-        walletKey,
-        mnemonic,
-        mnemonicKey
-      })
-    )
+    walletToSave = paratii.eth.wallet.encrypt(DEFAULT_PASSWORD)
   }
+  dispatch(
```

Comment:
The changes here boil down to this:

Whether we find a valid wallet in local storage or create a new one, we dispatch an action that will save that wallet in the `redux` store